src/heap/tst/HeapApiFunctionalityTest.cpp

```diff
@@ -420,11 +428,23 @@ TEST_F(HeapApiFunctionalityTest, AivHeapMinBlockFitResize)
     minBlockFitAllocResize(pHeap);
 }
 
+TEST_F(HeapApiFunctionalityTest, AivHeapUnalignedHeapLimit)
+{
+    PHeap pHeap;
+
+    EXPECT_TRUE(STATUS_SUCCEEDED(heapInitialize(MIN_HEAP_SIZE + 1, 20, FLAGS_USE_AIV_HEAP, &pHeap)));
+#ifdef ALIGNED_MEMORY_MODEL
+    EXPECT_EQ(MIN_HEAP_SIZE + 8, pHeap->heapLimit);
+#else
+    EXPECT_EQ(MIN_HEAP_SIZE + 1, pHeap->heapLimit);
+#endif
+}
+
 TEST_F(HeapApiFunctionalityTest, AivHeapResizeEdgeCases)
 {
     PHeap pHeap;
     ALLOCATION_HANDLE handle = INVALID_ALLOCATION_HANDLE_VALUE;
-    UINT32 size = MIN_HEAP_SIZE / NUM_ITERATIONS;
+    UINT32 size = ROUND_UP(MIN_HEAP_SIZE / NUM_ITERATIONS, 8);
```

Comment:
Would it be better to use HEAP_PACKED_SIZE?

Reply:
I think it's a good idea - though, we still would fail if the heap and the tests are compiled differently. Will change